shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -352,12 +348,6 @@ Engine::Engine(Delegate& delegate,
 
 Engine::~Engine() {
   shell_.reset();
-  for (const auto& thread : threads_) {
-    thread->Quit();
-  }
-  for (const auto& thread : threads_) {
-    thread->Join();
-  }
```

Comment:
Does FML do this automatically?

I suspect this was done to fix some issue that @cbracken or @chinmaygarde might know more about.

Reply:
It should do. The `ThreadHost` holds the various `fml::Thread` objects as `std::unique_ptr`s which will get destroyed when the `ThreadHost` goes out of scope, and in their dtor they call `Thread::Join()` which in turn terminates the `MessageLoop`: https://github.com/flutter/engine/blob/master/fml/thread.cc#L41

Reply:
I'm curious about the call to `Quit` - is posting `Terminate` in `Join` in FML equivalent?

Reply:
Yes. `flutter_runner::Thread::Quit()` calls `async::Loop::Quit()`. Calling `MessageLoop::Terminate()` eventually calls `MessageLoopFuchsia::Terminate()` which calls `async::Loop::Quit()`